### PR TITLE
[FEATURE] Allow plugins to declare their custom scanner or analyzer

### PR DESCRIPTION
**Pull request by ohader**
Fix: #5882

### CODE REVIEW

**Comment by ohader** [comment]
* Psalm self-test will fail, since new methods are not used - I was reluctant modifying existing `RegistrationInterface` - maybe there might be a new `FileTypeRegistrationInterface` (not sure about the name)
* documentation is missing - will take care
* probably some test cases would be nice as well

**Comment by muglug** [comment]
> Psalm self-test will fail, since new methods are not used

That will be fixed if you add tests!

**Comment by weirdan** [comment]
> I was reluctant modifying existing `RegistrationInterface`

It should be fairly safe to expand that. Third-party code is not expected to implement that interface, plugin authors should only consume it.

**Comment by ohader** [comment]
> > I was reluctant modifying existing `RegistrationInterface`
> 
> It should be fairly safe to expand that. Third-party code is not expected to implement that interface, plugin authors should only consume it.

Nice, just spotted this comment. Will extend the interface later...

File: .gitignore [hunk 1]
```diff
@@ -11,5 +11,4 @@
 /vendor-bin/*/vendor/
 /tests/fixtures/symlinktest/*
 
-.idea/inspectionProfiles/Project_Default.xml
-.idea/workspace.xml
+.idea/
```
Comment (ohader) [inline .gitignore:14]: Sorry for the drive-by fix...

**Comment by muglug** [comment]
Thanks!
